<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.149">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Ancient DNA Data Analysis - Workshop Uni Siegen 2022 - Exercise 2.1: Abba Baba test</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Ancient DNA Data Analysis - Workshop Uni Siegen 2022</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html">About</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#genotype-data" id="toc-genotype-data" class="nav-link active" data-scroll-target="#genotype-data">Genotype data</a></li>
  <li><a href="#exploring-neanderthal-human-genetic-similarity" id="toc-exploring-neanderthal-human-genetic-similarity" class="nav-link" data-scroll-target="#exploring-neanderthal-human-genetic-similarity">Exploring Neanderthal-human genetic similarity</a></li>
  <li><a href="#the-abba-baba-test" id="toc-the-abba-baba-test" class="nav-link" data-scroll-target="#the-abba-baba-test">The ABBA-BABA test</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Exercise 2.1: Abba Baba test</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="genotype-data" class="level2">
<h2 class="anchored" data-anchor-id="genotype-data">Genotype data</h2>
<div class="cell">

</div>
<p>A genotype matrix looks conceptually like this:</p>
<pre><code>SNPs / Individuals -&gt; Ind1 Ind2 Ind3 Ind4 ...
 |
 V
Snp1                   G    A    A    G   . . .
Snp2                   G    T    T    C   . . .
Snp3                   T    A    T    G   . . .
Snp4                   G    A    A    C   . . .
 .                     .    .    .    .   .
 .                     .    .    .    .     .
 .                     .    .    .    .       .</code></pre>
<p>We typically deal with dozens to tens of thousands of individuals, and hundreds of thousands to several million SNPs. In practice, to make this matrix easier computable, it is divided into three files:</p>
<ul>
<li>A file denoting the list of individuals (i.e.&nbsp;the column names)</li>
<li>A file denoting the list of SNPs (i.e.&nbsp;the row names)</li>
<li>A genotype matrix, containing the actual genotype data.</li>
</ul>
<p>Let’s take a look at the individual file first:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> data/Abba_Baba_dataset/Abba_Baba_dataset.ind</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NA12878.SG  F   CEU.SG
NA19240.SG  F   YRI.SG
VindijaG1_final_provisional.SG  F   VindijaG1_Neanderthal.SG
Chimp.REF   M   Chimp.REF</code></pre>
</div>
</div>
<p>where you see four individuals. In each row, the first column denotes the sample name, the second column the sex, and the third column a group label, which is irrelevant for now. Here is a quick summary for the four:</p>
<ul>
<li><code>NA12878.SG</code> is a present-day human from the US (with European ancestry).</li>
<li><code>NA19240.SG</code> is a present-day human from Nigeria.</li>
<li><code>VindijaG1_final_provisional.SG</code> is a Neanderthaler from Vindija Cave in Croatia, who died approximately 50,000 years ago.</li>
<li><code>Chimp.REF</code> is the Chimpanzee reference genome.</li>
</ul>
<p>Next, let’s view the first few SNPs:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> data/Abba_Baba_dataset/Abba_Baba_dataset.snp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>rs3094315   1   2.013e-2    752566  G   A
rs12124819  1   2.0242e-2   776546  G   A
rs28765502  1   2.2137e-2   832918  C   T
rs7419119   1   2.2518e-2   842013  G   T
rs950122    1   2.272e-2    846864  C   G
rs113171913 1   2.3436e-2   869303  T   C
rs13302957  1   2.4116e-2   891021  G   A
rs59986066  1   2.4183e-2   893462  C   T
rs112905931 1   2.426e-2    896271  T   C
rs6696609   1   2.4457e-2   903426  T   C</code></pre>
</div>
</div>
<p>Each row contains a SNP, with an ID (first column), chromosome (second column), position in base-pairs along the chromosome (fourth column), and the two alleles present at this SNP (last two columns).</p>
<p>Let’s see how many SNPs there are:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> data/Abba_Baba_dataset/Abba_Baba_dataset.snp <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> 1233013</code></pre>
</div>
</div>
<p>So around 1.2 million SNPs.</p>
<p>Let’s now take a look at the genotype file:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> data/Abba_Baba_dataset/Abba_Baba_dataset.geno</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>0009
0099
0299
0009
0090
0000
0092
9999
9909
0009</code></pre>
</div>
</div>
<p>Here, we have genotype data encoded in the EIGENSTRAT format, which - in our case - contains the following code:</p>
<ul>
<li><code>0</code> -&gt; Allele 1 (the “alternative allele”)</li>
<li><code>2</code> -&gt; Allele 2 (the “reference allele”)</li>
<li><code>9</code> -&gt; Missing data</li>
</ul>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Technically, the format also contains <code>1</code> for heterozygous genotypes, but the data we analyse here is “pseudo-haploid”, so represents only one allele for each individual.</p>
</div>
</div>
<p>Exercise: Obviously, the genotype file contains the right number of columns (four) for the four individuals we have. Check that it also contains the right number of rows, i.e.&nbsp;the same number that you obtained above using the <code>wc</code> command.</p>
</section>
<section id="exploring-neanderthal-human-genetic-similarity" class="level2">
<h2 class="anchored" data-anchor-id="exploring-neanderthal-human-genetic-similarity">Exploring Neanderthal-human genetic similarity</h2>
<p>Given that we now have a large number of genotypes at our hands, between Neanderthals and modern humans, we can ask a very simple question: Is the Neanderthal genome genetically more similar to one of the two modern humans than the other?</p>
<p>To answer this, we need to count in how many positions the Neanderthal matches each of the two modern human genomes. And to make this more meaningful, we focus on genetic variants which are different from Chimpanzee, i.e.&nbsp;mutations which appeared along the human lineage since the split from Chimpanzee.</p>
<p>We first select from the genotype file the relevant columns. For example, to select genotypes from the US-individual (first column), Neanderthal (third column) and Chimnpanzee (fourth column) we can use the command <code>cut -c1,3,4</code>, like this:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-c1,3,4</span> data/Abba_Baba_dataset/Abba_Baba_dataset.geno <span class="kw">|</span> <span class="fu">head</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>009
099
099
009
090
000
092
999
909
009</code></pre>
</div>
</div>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Piping a result of a computation into <code>head</code> is a very common trick in exploratory data analysis in the shell. It avoids a super-long output. Here, we just would like to see the first ten lines of the result to see whether it’s working as expected.</p>
</div>
</div>
<p>OK, this looks good. Now we have to filter lines at which the two genotypes in Neanderthal and Human are similar, but different from Chimpanzee. In other words, we are looking for two possible patterns in the three individuals: <code>002</code> or <code>220</code>. We use the command <code>grep</code>, and then count the lines, using <code>wc</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-c1,3,4</span> data/Abba_Baba_dataset/Abba_Baba_dataset.geno <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-e</span> 002 <span class="at">-e</span> 220 <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   65283</code></pre>
</div>
</div>
<p>Exercise: Repeat the same exercise for the Nigerian individual (which is in column 2 instead of column 1 of the genotype file). You should get <code>61933</code>.</p>
<p>Comparing the two numbers, we see that the US individual shares around 3000 more variants with the Croatian Neanderthal than the Nigerian individual does (<code>65283</code> vs <code>61933</code>).</p>
</section>
<section id="the-abba-baba-test" class="level2">
<h2 class="anchored" data-anchor-id="the-abba-baba-test">The ABBA-BABA test</h2>
<p>The analysis that we did above is very similar to what has been called the ABBA-BABA test in the population genetic literature (<span class="citation" data-cites="Green2010-rv">Green et al. (<a href="#ref-Green2010-rv" role="doc-biblioref">2010</a>)</span>). Specifically, when considering the quadruple of individuals as above, in the order:</p>
<ul>
<li>Outgroup (such as Chimp)</li>
<li>Neanderthal</li>
<li>Individual with Non-African ancestry</li>
<li>Individual with African ancestry</li>
</ul>
<p>we consider two types of allelic patterns, ABBA and BABA. The four letters refer to the alleles in the four individuals, and A and B simply denote the two alleles. The two patterns correspond to the above computed similarities of derived variants from Chimp:</p>
<ul>
<li>ABBA -&gt; Here, the Non-African, but not the African, shares the derived allele with Neanderthal</li>
<li>BABA -&gt; Here, the African, but not the Non-African, shares the derived allele with Neanderthal.</li>
</ul>
<p><span class="citation" data-cites="Green2010-rv">Green et al. (<a href="#ref-Green2010-rv" role="doc-biblioref">2010</a>)</span> then defined a simple statistic:</p>
<p><span class="math display">\[
D = \frac{n_{ABBA} - n_{BABA}}{n_{ABBA} + n_{BABA}}.
\]</span></p>
<p>Exercise: Above we have computed counts of the slightly simpler pattern <code>AAB</code> in their two realisations <code>002</code> and <code>220</code>. To compute the D-Statistic, we need to operate on all four individuals, where <code>ABBA</code> is realised by <code>0220</code> and <code>2002</code>, and <code>BABA</code> is realised by <code>2020</code> and <code>0202</code>. Compute the count of <code>ABBA</code> and <code>BABA</code> using the above <code>grep</code> pipeline. You should get <span class="math inline">\(n_{ABBA} = 21787\)</span> and <span class="math inline">\(n_{BABA} = 20250\)</span>.</p>
<p>Exercise: Open an R console to compute the D-Statistic itself, using the two numbers. The number you get is a (very rough) estimate of the admixture proportion of Neanderthal ancestry in modern humans with Non-African ancestry today.</p>



</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-Green2010-rv" class="csl-entry" role="doc-biblioentry">
Green, Richard E, Johannes Krause, Adrian W Briggs, Tomislav Maricic, Udo Stenzel, Martin Kircher, Nick Patterson, et al. 2010. <span>“A Draft Sequence of the Neandertal Genome.”</span> <em>Science</em> 328 (5979): 710–22. <a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;id=20448178&amp;retmode=ref&amp;cmd=prlinks">http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;id=20448178&amp;retmode=ref&amp;cmd=prlinks</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>