---
title: "Part 1 - From bones to sequences"
subtitle: "Workshop at the University of Siegen and Bonn, Sept 2022"
author: Stephan Schiffels, MPI for Evolutionary Anthropology, Leipzig
format: revealjs
engine: knitr
---

# Introduction

## What is Archaeogenetics?

-   Genetic Analysis of prehistoric organic remains
-   Examples: bones, teeth, dental calculus, antler, hair, seed, wood, pollen, soil, coprolites

![](images/part_01/adna_sources_1.png) ![](images/part_01/adna_sources_2.png)

## Why archaeogenetics? {.smaller}

-   Why are we doing this? Why are prehistoric genetic data important?
-   Because genetic variability in the past can not be derived unambiguously from genetic variability in the presence.

::: r-stack
![](images/part_01/phylotree_schematic_empty.png){.fragment .fade-in-then-out width="800"}

![](images/part_01/phylotree_schematic_skulls.png){.fragment width="800"}
:::

::: aside
Images by Wolfgang Haak
:::

## Sources of ancient DNA

::: columns
::: {.column width="40%"}
::: {.fragment fragment-index="1"}
-   Human DNA (Mitochondrial DNA, Nuclear DNA)
:::

::: {.fragment fragment-index="2"}
-   Pathogen DNA (Bacteria, Viruses, Protists)
:::

::: {.fragment fragment-index="3"}
-   Environmental DNA (Plants, Funghi, Bacteria, Viruses, Animals)
:::
:::

::: {.column width="60%"}
![](images/part_01/cell.png){.fragment fragment-index="1"}

![](images/part_01/pathogens.png){.fragment with="200" fragment-index="2"} ![](images/part_01/environ_dna.png){.fragment width="300" fragment-index="3"}
:::
:::

## What happens in an ancient DNA laboratory?

![](images/part_01/adna_lab_schematic.png){fig-align="center"}

## From evolutionary to family history {.smaller}

![](images/part_01/popgen_scales_novembre2011.png){fig-align="center" height="520"}

::: aside
Novembre and Ramachandran 2011, Annu. Rev. Genomics Hum. Genet. 2011.
:::

## DNA degradation {.smaller}

::: columns
::: {.column width="60%"}
-   Ancient DNA is **short** (50-100 bp)
-   Ancient DNA has **chemically changed** after death (C-\>U-\>T changes at the ends)
-   Ancient DNA is partly **single-stranded** (strand breaks)

![](images/part_01/strand_breaks.png)
:::

::: {.column width="40%"}
![](images/part_01/adna_size_dist.png)

![](images/part_01/damage_plot.png)

::: aside
Images: Stoneking and Krause 2011, Nat Rev Gen; Christina Warinner
:::
:::
:::

# Sequencing Data {.smaller}

This is how sequencing reads look like on a computer:

```{bash}
#| echo: true
samtools view data/Sample.bam | cut -f10 | head
```

* Problem: Sequencing reads are **short**, the human genome is **long**.

::: callout-note
Note that the `samtools view` command is used here and introduced later.
:::

::: callout-tip
If you're familiar with the command line, you can try out the command (`samtools view ...`) yourself. The data is available from this repository.
:::

## Solution: Alignments {.smaller}

* Every short fragment gets _mapped_ onto the human genome
* Fast mapping algorithms use sophisticated indexing of the human genome to quickly find a matching position

:::: r-stack

::: {.fragment .fade-in-then-out}
A typical way to visualise such alignments is shown here:
```{bash}
samtools tview data/Sample.bam -d t --reference data/hg19_MT.fasta
```
:::

::: fragment
Or without the reference genome:
```{bash}
samtools tview data/Sample.bam -d t
```
:::

::::

::: aside
Top line: Reference genome

Second line: Consensus sequence

Other lines: fragments
:::

## Interlude - RStudio Cloud {.smaller}

* RStudio is an open source and free platform for data science
* RStudio-Cloud is a browser-based version
![](images/part_01/rstudio-cloud.png)

## `r fontawesome::fa("laptop-code", "white")` Your Turn (Exercise 1.1) {background-color="#447099" transition="slide-in" .smaller}

- Make sure you have access to our RStudio project, as described on the workshop webpage (`https://stschiff.github.io/Ancient_DNA_Workshop_Sept_2022/`)
- Open the file `exercise_1_1_alignment.txt` under `presentations`. 
- Edit the file to visually align the sequencing Reads to the given Reference genome:


    Reference_Genome: 
    AACACTTGGTCACTGGGGTTTACACTGGTACACTGGACTACCCTGGGTACACTGACTACGACTACGACTACGTTTACCCCAATTT

    Sequencing_Reads:     
    GTTTACACTGGTGCACTGGACTACCCTG
    ACACTGGTGCACTGGACTACCCTGGGTACACTGACTAC
    TGGTGCACTGGACTACCCTGGGTACACT
    GACTACCCTGGGTACACTGACTACGA
    GGTACACTGACTACGACTACGACTACGTT


::: callout-tip
There may be single basepair mismatches. Find them!
:::

## Solution 1.1


                            Mismatch A -> G               Mismatch G->C
                                 |                             |
                                 |                             |
                                 V                             V
    AACACTTGGTCACTGGGGTTTACACTGGTACACTGGACTACCCTGGGTACACTGACTACGACTACGACTACGTTTACCCCAATTT
                     GTTTACACTGGTGCACTGGACTACCCTG
                         ACACTGGTGCACTGGACTACCCTGGGTACACTGACTAC
                             TGGTGCACTGGACTACCCTGGGTACACT
                                       GACTACCCTGGGTACACTGACTACCA
                                                 GGTACACTGACTACCACTACGACTACGTT


## `r fontawesome::fa("laptop-code", "white")` Your Turn (Exercise 1.2) {background-color="#447099" transition="slide-in" .smaller}

- Click into the terminal in RStudio
- Change into the `presentations` directory by typing `cd presentations`, followed by `<Enter>`
- Run `samtools tview data/Sample.bam`
- Also try `samtools tview data/Sample.bam --reference data/HG19_MT.fasta`

::: callout-tip
- Use your cursors to move left and right
- Use space or Ctrl-L to make bigger jumps to the right
- Use or Ctrl-H for bigger jumps to the left.
- Type `?` to get a quick help
- Type `q` to quit.
:::

## SNP calling {auto-animate="true"}

<pre>
...ACCCAAACTTTGCGCATATCGCTCTGGTTTAAACACCACACCGGGTTTACATCACTCG...
         ACTTTGCGC<span style="color:red">G</span>TAT             ACACCACACCGGGTTT
             TGCGC<span style="color:red">G</span>TATCGCT              ACACCGGGTTTACATCA
                  <span style="color:red">G</span>TATCGCTC<span style="color:red">A</span>GGTTTA          CGGGCTTACATCACTCG
                      CGCTC<span style="color:red">A</span>GGTTTAAACAC        GCTTACATCACTCG...

</pre>
 
<pre>
...ACCCAAACTTTGCGC<span style="color:red">G</span>TATCGCTC<span style="color:red">A</span>GGTTTAAACACCACACCGGGTTTACATCACTCG...
</pre>

<pre>
...ACCCAAACTTTGCGCATATCGCTCTGGTTTAAACACCACACCGGGTTTACATCACTCG...
    CCCAAACTTTGCGCAT<span style="color:red">T</span>T    CTG<span style="color:red">C</span>TTTAAACACCA    GGGTTTACATCACT   
             TGCGCAT<span style="color:red">T</span>TCGCTCTG<span style="color:red">C</span>TTT
                  AT<span style="color:red">T</span>TCGCTCTG<span style="color:red">C</span>TT

</pre>

<pre>
...ACCCAAACTTTGCGCAT<span style="color:red">T</span>TCGCTCTG<span style="color:red">C</span>TTTAAACACCACACCGGGTTTACATCACTCG...
</pre>

## Next Slide {auto-animate="true"}

<pre>
...ACCCAAACTTTGCGC<span style="color:red">G</span>TATCGCTC<span style="color:red">A</span>GGTTTAAACACCACACCGGGTTTACATCACTCG...
</pre>

<pre>
...ACCCAAACTTTGCGCAT<span style="color:red">T</span>TCGCTCTG<span style="color:red">C</span>TTTAAACACCACACCGGGTTTACATCACTCG...
</pre>
